x/blob/ante/max_total_blob_size_ante.go

```diff
@@ -20,36 +20,33 @@ func NewMaxBlobSizeDecorator(k BlobKeeper) MaxTotalBlobSizeDecorator {
 }
 
 // AnteHandle implements the Cosmos SDK AnteHandler function signature. It
-// returns an error if tx contains a MsgPayForBlobs where the total blob size is
-// greater than the max total blob size.
+// returns an error if tx contains a MsgPayForBlobs where the blob sizes would
+// occupy more shares than the largest possible data square.
 func (d MaxTotalBlobSizeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
 	if !ctx.IsCheckTx() {
 		return next(ctx, tx, simulate)
 	}
 
-	max := d.maxTotalBlobSize(ctx)
+	maxBlobShares := d.getMaxBlobShares(ctx)
 	for _, m := range tx.GetMsgs() {
 		if pfb, ok := m.(*blobtypes.MsgPayForBlobs); ok {
-			if total := getTotal(pfb.BlobSizes); total > max {
-				return ctx, errors.Wrapf(blobtypes.ErrTotalBlobSizeTooLarge, "total blob size %d exceeds max %d", total, max)
+			if sharesNeeded := getSharesNeeded(pfb.BlobSizes); sharesNeeded > maxBlobShares {
+				return ctx, errors.Wrapf(blobtypes.ErrTotalBlobSizeTooLarge, "shares needed %d exceeds max blob shares %d", sharesNeeded, maxBlobShares)
 			}
 		}
 	}
 
 	return next(ctx, tx, simulate)
 }
 
-// maxTotalBlobSize returns the max the number of bytes available for blobs in a
-// data square based on the max square size. Note it is possible that txs with a
-// total blob size less than this max still fail to be included in a block due
-// to overhead from the PFB tx and/or padding shares.
-func (d MaxTotalBlobSizeDecorator) maxTotalBlobSize(ctx sdk.Context) int {
+// getMaxBlobShares returns the max the number of shares available for blob data.
+func (d MaxTotalBlobSizeDecorator) getMaxBlobShares(ctx sdk.Context) int {
 	squareSize := d.getMaxSquareSize(ctx)
 	totalShares := squareSize * squareSize
 	// The PFB tx share must occupy at least one share so the # of blob shares
 	// is at least one less than totalShares.
 	blobShares := totalShares - 1
```

Comment:
Suggestion: do we want to bump this from 1 to 2 to have one intranamespace padding as buffer

I just think it's better to be conservative about this

Reply:
Yea I can see the argument to account for padding as buffer. But how much padding do we account for?
- The square could contain one or more [namespace padding](https://celestiaorg.github.io/celestia-app/specs/shares.html#namespace-padding-share) 
- The square could contain one or more [primary reserved padding](https://celestiaorg.github.io/celestia-app/specs/shares.html#primary-reserved-padding-share)

My concern is that if this ante handler accounted for worst case padding then it may reject blobs from inclusion into the mempool even though that tx could be included in a square.

One idea for a more sophisticated ante handler is to construct a data square with just this one tx and verify that the # of shares used is < maxSquareSize * maxSquareSize

Reply:
> My concern is that if this ante handler accounted for worst case padding then it may reject blobs from inclusion into the mempool even though that tx could be included in a square.

That's probably okay if we're talking about a blob that is still 64 * 64 shares i.e. it's a pretty big blob. It's probably better that the user gets the feedback immediately via `CheckTx` then having their transaction mysteriously unable to get on-chain